Source/GMCAbilitySystem/Private/Components/GMCAbilityComponent.cpp

```diff
@@ -22,6 +22,45 @@ UGMC_AbilitySystemComponent::UGMC_AbilitySystemComponent(const FObjectInitialize
 	SetIsReplicatedByDefault(true);
 }
 
+FDelegateHandle UGMC_AbilitySystemComponent::AddFilteredTagChangeDelegate(const FGameplayTagContainer& Tags,
+	const FGameplayTagFilteredMulticastDelegate::FDelegate& Delegate)
+{
+	TPair<FGameplayTagContainer, FGameplayTagFilteredMulticastDelegate>* MatchedPair = nullptr;
+
+	for (auto& SearchPair : FilteredTagDelegates)
+	{
+		if (SearchPair.Key == Tags)
+		{
+			MatchedPair = &SearchPair;
+		}
+	}
+
+	if (!MatchedPair)
+	{
+		MatchedPair = new(FilteredTagDelegates) TPair<FGameplayTagContainer, FGameplayTagFilteredMulticastDelegate>(Tags, FGameplayTagFilteredMulticastDelegate());
+	}
+
+	return MatchedPair->Value.Add(Delegate);
+}
+
+void UGMC_AbilitySystemComponent::RemoveFilteredTagChangeDelegate(const FGameplayTagContainer& Tags,
+	FDelegateHandle Handle)
+{
+	for (int32 Index = 0; Index < FilteredTagDelegates.Num(); Index++)
+	{
+		TPair<FGameplayTagContainer, FGameplayTagFilteredMulticastDelegate>& SearchPair = FilteredTagDelegates[Index];
+		if (SearchPair.Key == Tags)
+		{
+			SearchPair.Value.Remove(Handle);
+			if (!SearchPair.Value.IsBound())
+			{
+				FilteredTagDelegates.RemoveAt(Index);
```

Comment:
Pretty sure this will cause an error with the iteration, as removing an element during the loop will change the array's indexes mid-loop

Reply:
You are correct; I had changed it to reverse iterator in my game-specific branch (which has other changes), and forgot to do it here. Changed!